hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/domain/ContractResultServiceImpl.java

```diff
@@ -88,13 +89,14 @@ public void process(@NonNull RecordItem recordItem, Transaction transaction) {
 
         // contractResult
         var transactionHandler = transactionHandlerFactory.get(TransactionType.of(transaction.getType()));
-
+        var throwRecoverableError = shouldThrowRecoverableError(transactionRecord);
         // in pre-compile case transaction is not a contract type and entityId will be of a different type
         var contractId = (contractCallOrCreate
                         ? Optional.ofNullable(transaction.getEntityId())
-                        : entityIdService.lookup(functionResult.getContractID()))
+                        : entityIdService.lookup(functionResult.getContractID(), throwRecoverableError))
                 .orElse(EntityId.EMPTY);
-        var isRecoverableError = EntityId.isEmpty(contractId)
+        var isRecoverableError = throwRecoverableError
```

Comment:
While probably they're both guaranteed to be set the same, just in case we should separate recoverable error logic. That is, don't call shouldThrowRecoverableError() once and pass it to both but check them separately with the proper field.

```suggestion
        var isRecoverableError = !(recordItem.isSuccessful() && functionResult.getContractID().hasEvmAddress())
```

Reply:
I tried a few things here and my changes broke the unit tests, will continue to look at this.

Reply:
Thanks I have now fixed this, please re-review.